scripts/quests/bastok/Chips.lua

```diff
@@ -0,0 +1,122 @@
+-----------------------------------
+-- Chips
+-----------------------------------
+-- Log ID: 1, Quest ID: 82
+-----------------------------------
+local metalworksID = zones[xi.zone.METALWORKS]
+-----------------------------------
+
+local quest = Quest:new(xi.questLog.BASTOK, xi.quest.id.bastok.CHIPS)
+
+quest.reward =
+{
+    item = xi.item.CCB_POLYMER_PUMP,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_AVAILABLE and
+            (player:getCurrentMission(xi.mission.log_id.COP) > xi.mission.id.cop.ONE_TO_BE_FEARED or
+            (player:getCurrentMission(xi.mission.log_id.COP) == xi.mission.id.cop.ONE_TO_BE_FEARED and
+            player:getCharVar('Mission[6][638]Status') >= 1)) and
+            not quest:getMustZone(player)
+        end,
+
+        [xi.zone.LOWER_JEUNO] =
+        {
+            ['Ghebi_Damomohe'] = quest:progressEvent(169),
+
+            onEventFinish =
+            {
+                [169] = function(player, csid, option, npc)
+                    if option == 0 then
+                        quest:begin(player)
+                    else quest:setMustZone(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.METALWORKS] =
+        {
+            ['Cid'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if
+                        npcUtil.tradeHasExactly(trade, {
+                        xi.item.CARMINE_CHIP,
```

Comment:
indent needed for subparams on multiple lines, an alternative would be to define the table elsewhere